### PR TITLE
Fixed storage args, added "max-connections" arg to `blob download-batch`

### PR DESCRIPTION
**Pull request by williexu**
---
Fixes: https://github.com/Azure/azure-cli/issues/5308
-fixed instances of `--max-connections` as well
-added `--max-connections` arg to `storage blob download-batch` to be consistent with other commands
-small fix for failing live tests, and added test coverage 

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5384
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-storage/HISTORY.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ Release History
 * Added progress reporting for all upload/download commands, including batch.
 * `storage account check-name`: fixed bug preventing "-n" arg option.
 * Added 'snapshot' column to table output for blob list/show.
+* Fixed bugs with various parameters that needed to be parsed as ints, added test coverage.
+* Small fix with test, `storage blob service-properties show`: "hourMetrics.enabled" defaults to false.
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/HISTORY.rst]: If this is a service change, make sure we understand it and it is documented.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/blob.py [hunk 2]
```diff
@@ -99,15 +99,15 @@ def action_file_copy(file_info):
 
 # pylint: disable=unused-argument
 def storage_blob_download_batch(client, source, destination, source_container_name, pattern=None, dryrun=False,
-                                progress_callback=None):
+                                progress_callback=None, max_connections=2):
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/blob.py]: Can we defer the decision of default value to the SDK?

Comment (williexu) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/blob.py]: yes indeed

**Comment by tjprescott** [comment]
@williexu feel free to merge when you are satisfied

**Comment by williexu** [comment]
The service team is now aware of the recent "hourMetrics.enabled" default change. If they make a fix to restore the previous behavior, I'll edit the test as needed when they do. 
For now, will merge this once CI passes, as this PR fixes a bunch of commands and addresses the current service behavior.

**Pull request merged**